saya/core/src/storage/sql_lite/mod.rs

```diff
@@ -0,0 +1,141 @@
+use std::fs;
+use std::path::Path;
+use std::time::Duration;
+
+use anyhow::Error;
+use log::trace;
+use sqlx::query;
+use sqlx::sqlite::SqlitePoolOptions;
+use sqlx::Pool;
+use sqlx::Sqlite;
+
+mod storage;
+mod utils;
+#[derive(Clone)]
+pub struct SqliteDb {
+    pub(crate) pool: Pool<Sqlite>,
+}
+
+impl SqliteDb {
+    pub async fn new(path: &str) -> Result<Self, Error> {
+        // Check if there is a database file at the path
+        if !Path::new(path).try_exists()? {
+            trace!(
+                "Database file not found. A new one will be created at: {}",
+                path
+            );
+            fs::File::create(path)?;
+        } else {
+            trace!("Database file found at: {}", path);
+        }
+
+        let pool = SqlitePoolOptions::new()
+            .acquire_timeout(Duration::from_secs(10))
+            .max_connections(50)
+            .connect(&format!("sqlite:{}", path))
+            .await?;
+
+        let table_exists = Self::check_table_exists(&pool).await?;
```

Comment:
Usually to follow up a DB structure that can be backward compatible and support changes, migrations are used to describe each changes over the life of the DB.

Having migrations managed from SQL code instead of having the rust code describing the tables, may make the process easier to maintain the DB over time.

For now we may consider that Saya DB is relatively short living, and may add this in a future rework.